{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "3fd84f27-e199-40a6-a172-f3315e1d4bbf",
   "metadata": {},
   "source": [
    "# 1. Locating the TP53 Variant using the Variant ID rs11540652 from Ensembl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "528a2360-1083-4f2d-a19a-41a83ad3d9ba",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting the location of tp53 variant id rs11540652\n",
      "Variant rs11540652 location: 17:7674220-7674220\n"
     ]
    }
   ],
   "source": [
    "!echo \"Getting the location of tp53 variant id rs11540652\" \n",
    "!curl -s 'https://rest.ensembl.org/variation/human' -H 'Content-type:application/json' \\\n",
    "-H 'Accept:application/json' -X POST -d '{ \"ids\" : [\"rs11540652\"] }' | jq -r '\"Variant rs11540652 location: \\(.rs11540652.mappings[0].location)\"'"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c5bc1845-daaf-42cc-86b3-31a856c9feea",
   "metadata": {},
   "source": [
    "# 2. Query Ensembl VEP (Variant Effect Predictor) API for TP53 Variant (c.743G>A)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "3067fe14-d57a-45eb-bcc3-509032edce02",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Getting more detail on variant location and mutation type classification\n",
      "id: ENST00000269305.9:c.743G>A\n",
      "variant_class: SNV\n",
      "input: ENST00000269305.9:c.743G>A\n",
      "transcript_id: ENST00000269305\n",
      "amino_acids: R/Q\n",
      "hgvsc: ENST00000269305.9:c.743G>A\n",
      "strand: -1\n",
      "variant_allele: A\n",
      "codons: cGg/cAg\n",
      "exon: 7/11\n",
      "hgvsp: ENSP00000269305.4:p.Arg248Gln\n",
      "consequence_terms: [\"missense_variant\"]\n",
      "gene_id: ENSG00000141510\n",
      "gene_symbol: TP53\n",
      "biotype: protein_coding\n",
      "start: 7674220\n",
      "end: 7674220\n"
     ]
    }
   ],
   "source": [
    "!echo \"Getting more detail on variant location and mutation type classification\" \n",
    "!curl -s 'https://rest.ensembl.org/vep/human/hgvs' -H 'Content-type:application/json' \\\n",
    "-H 'Accept:application/json' -X POST -d  '{\"hgvs_notations\": [\"ENST00000269305.9:c.743G>A\"], \"numbers\": 1, \"hgvs\": 1, \\\n",
    "\"transcript_id\": \"ENST00000269305\", \\\n",
    "\"variant_class\": 1}' | \\\n",
    "jq -r '.[] | . as $root | .transcript_consequences[] | \"id: \\($root.id)\", \"variant_class: \\($root.variant_class)\", \"input: \\($root.input)\", \\\n",
    "\"transcript_id: \\(.transcript_id)\", \"amino_acids: \\(.amino_acids)\", \"hgvsc: \\(.hgvsc)\", \"strand: \\(.strand)\", \"variant_allele: \\(.variant_allele)\", \\\n",
    "\"codons: \\(.codons)\", \"exon: \\(.exon)\", \"hgvsp: \\(.hgvsp)\", \"consequence_terms: \\(.consequence_terms)\", \"gene_id: \\(.gene_id)\", \\\n",
    "\"gene_symbol: \\(.gene_symbol)\", \"biotype: \\(.biotype)\", \"start: \\($root.start)\", \"end: \\($root.end)\" '"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.19"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
